sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/util/BuilderHelper.java

```diff
@@ -233,4 +238,20 @@ private static Tracer createTracer(ClientOptions clientOptions) {
         return TracerProvider.getDefaultProvider()
             .createTracer(CLIENT_NAME, CLIENT_VERSION, STORAGE_TRACING_NAMESPACE_VALUE, tracingOptions);
     }
+
+    public static Context addUpnHeader(Supplier<Boolean> upnHeaderValue, Context context) {
+        Boolean value = upnHeaderValue.get();
+        if (value == null) {
+            return context;
+        }
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("x-ms-upn", Boolean.toString(value));
```

Comment:
nit: `set(String, String)` is deprecated and `set(HttpHeaderName, String)` should be used instead.

I recommend creating a constant in this class `private static final HttpHeaderName X_MS_UPN = HttpHeaderName.fromString("x-ms-upn)` and changing this call to `headers.set(X_MS_UPN, Boolean.toString(value))`